### PR TITLE
add extra annotations

### PR DESCRIPTION
**Pull request by wansir**
Signed-off-by: hongming <talonwan@yunify.com>

### CODE REVIEW

File: pkg/models/resources/extraannotations.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	return item
+}
+
+func isPvcInUse(pods []*v1.Pod, pvcName string) bool {
+	for _, pod := range pods {
+		volumes := pod.Spec.Volumes
+		for _, volume := range volumes {
+			pvc := volume.PersistentVolumeClaim
+			if pvc != nil && pvc.ClaimName == pvcName {
+				return true
```
Comment (zryfish) [inline pkg/models/resources/extraannotations.go]: a pvc can be mounted by multiple pods, `kubectl describe pvc` can show pvc mounted by which pods. please have a look at this pr https://github.com/kubernetes/kubernetes/pull/65837/files

File: pkg/models/resources/resources.go [hunk 2]
```diff
@@ -142,9 +143,9 @@ func ListResources(namespace, resource string, conditions *params.Conditions, or
 		return nil, err
 	}
 
-	for i, d := range result {
+	for i, item := range result {
 		if i >= offset && (limit == -1 || len(items) < limit) {
-			items = append(items, d)
+			items = append(items, injector.addExtraAnnotations(item))
```
Comment (zryfish) [inline pkg/models/resources/resources.go]: I think it's better to move add annotations part to `persistentvolumeclaims.go`, for now, only pvc needed to be dealt with differently

File: pkg/models/resources/extraannotations.go (hunk 1 above)
Comment (wansir) [reply pkg/models/resources/extraannotations.go]: just a status flag, this API is enough
https://github.com/kubesphere/kubesphere/blob/3e8035dc2b2848afaf896bc01626ecd185302e1a/pkg/models/resources/pods.go#L165-L168

File: pkg/models/iam/im.go [hunk 3]
```diff
@@ -351,11 +351,11 @@ func ListUsers(conditions *params.Conditions, orderBy string, reverse bool, limi
 			j = tmp
 		}
 		switch orderBy {
-		case "username":
-			fallthrough
 		case "createTime":
 			return users[i].CreateTime.Before(users[j].CreateTime)
 		default:
+			fallthrough
+		case "username":
```
Comment (zryfish) [inline pkg/models/iam/im.go]: so `username` is the default case, 
```
default:
        return strings.Compare(users[i].Username, users[j].Username) <= 0
```

File: pkg/models/status/status.go [hunk 4]
```diff
@@ -34,13 +35,19 @@ func GetNamespacesResourceStatus(namespace string) (*workLoadStatus, error) {
...
+			notReadyStatus = resources.StatusFailed
+		default:
+			notReadyStatus = resources.StatusUpdating
 		}
 
-		notReadyList, err = resources.ListResources(namespace, resource, &params.Conditions{Match: map[string]string{"status": notReadyStatus}}, "", false, -1, 0)
+		notReadyList, err = resources.ListResources(namespace, resource, &params.Conditions{Match: map[string]string{resources.Status: notReadyStatus}}, "", false, -1, 0)
 
 		if err != nil {
 			return nil, err
```
Comment (zryfish) [inline pkg/models/status/status.go]: add log

**Pull request merged**